code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/ApiAuthenticator.java

```diff
@@ -150,7 +154,7 @@ protected Account getAccount(ApiRequest request) {
         for (IdentityTransformationHandler identityTransformationHandler : identityTransformationHandlers) {
             identities.addAll(identityTransformationHandler.getIdentities(account));
         }
-
+        identities.remove(null);
```

Comment:
I could but that would require like 7 null checks in different places. Sometimes it is an error for null identities and some times it is not. But in this case it is not, however we do not want nulls in the set.